server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/DistributionPlanner.java

```diff
@@ -94,6 +109,36 @@ public PlanNode visitDeviceMerge(TimeJoinNode node, DistributionPlanContext cont
       return null;
     }
 
+    @Override
+    public PlanNode visitMetaMerge(MetaMergeNode node, DistributionPlanContext context) {
```

Comment:
Here all the MetaScanNode will be linked into one MetaMergeNode although some of them belong to one SchemaRegion. And in the following ExchangeNodeAdder phase, there will be `N-1` ExchangeNodes are added, which may lead to too many FragmentInstances issue.

Maybe we can group the MetaScanNodes which belong to one SchemaRegion using a MetaMergeNode...

Reply:
> Here all the MetaScanNode will be linked into one MetaMergeNode although some of them belong to one SchemaRegion. And in the following ExchangeNodeAdder phase, there will be `N-1` ExchangeNodes are added, which may lead to too many FragmentInstances issue.
> 
> Maybe we can group the MetaScanNodes which belong to one SchemaRegion using a MetaMergeNode...

It seems that this comment is not correct... Currently each SchemaRegion will execute one MetaScanNode with same parameter, which is correct.

Another question is that is the `schemaRegions` here is the `minimum` collections calculated from PatialPath rather than all the SchemaRegions ?

Reply:
> Another question is that is the schemaRegions here is the minimum collections calculated from PatialPath rather than all the SchemaRegions ?

All the SchemaRegions here is calculated by `IPartitionFetcher#fetchSchemaPartitionInfo(String devicePath)`